server/blob/blob-deduplicating/src/test/scala/org/apache/james/server/blob/deduplication/GCPropertiesTest.scala

```diff
@@ -0,0 +1,216 @@
+/***************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.server.blob.deduplication
+
+import java.nio.charset.StandardCharsets
+
+import com.google.common.hash
+import org.apache.james.blob.api.BlobId
+import org.apache.james.server.blob.deduplication.Generators.{OnePassGCTestParameters, TestParameters}
+import org.scalacheck.Prop.forAll
+import org.scalacheck.Test.Parameters
+import org.scalacheck.{Arbitrary, Gen, Properties, Shrink}
+
+case class GenerationAwareBlobId(generation: Generation, hash: String) extends BlobId {
+  override def asString(): String = s"${generation}_$hash"
+}
+
+object Generators {
+
+  // generate a sequence of Generations with monotonic numeric ids
+  // 80% of the time, the generation id is not incremented
+  // 19% of the time, the generation id is incremented by 1
+  // 1% of the time, the generation id is incremented by 2
+  // i.e. (0, 0, 0, 2, 3, 3, 4, 5, 5, 5, 5)
```

Comment:
This comment is awesome!

I whish I had the same for every generators present here...

Reply:
Case by case why not, but it does not makes sense to do it systematically

Reply:
Yet it is very friendly for a reviewer neofit ;-)

Reply:
Sure, but I am not fan of literate programming, which actually makes the code hard to read.